docs/schema/index-management/index-lifecycle.md

```diff
@@ -26,28 +26,72 @@ mgmt.commit();
+:   The index has been created on the current JanusGraph instance, but it is not yet known to all instances in the cluster.
+    This state is not shown in the diagram above because it is not stable in terms of synchronization between instances.
 
--   **ENABLED** The index is enabled and in use
+**REGISTERED**
 
--   **DISABLED** The index is disabled and no longer in use
+:   The index is known to all instances in the cluster.
+    However, it is not in use yet.
+    In this state, the index contains no data and is not eligible to be used in queries.
 
-## Actions (SchemaAction)
-The following actions can be performed on an index to change its state via `mgmt.updateIndex()`:
+**ENABLED**
+
+:   The index is enabled and can be used to answer queries.
+    Furthermore, this is the only state in which insertions and deletions in the graph are forwarded to the index.
 
--   **REGISTER_INDEX** Registers the index with all instances in the graph cluster. After an index is installed, it must be registered with all graph instances
+**DISABLED**
 
--   **REINDEX** Re-builds the index from the graph
+:   The index is disabled and will not be used to answer queries.
+It also does no longer receive graph updates.
+Therefore, it is recommended to use `REINDEX` to re-enable the index.
 
--   **ENABLE_INDEX** Enables the index so that it can be used by the query processing engine. An index must be registered before it can be enabled.
+**DISCARDED**
 
--   **DISABLE_INDEX** Disables the index in the graph so that it is no longer used.
+:   The index has been decommissioned and its contents have been deleted.
+    The only remaining artifact is the schema vertex representing the index.
+
+## Actions (SchemaAction)
+The following actions can be performed on an index to change its state via `mgmt.updateIndex()`:
 
--   **REMOVE_INDEX** Removes the index from the graph (optional operation). Only on composite index.
+**REGISTER_INDEX**
+:   Registers the index with all instances in the graph cluster.
+After an index is installed, it must be registered with all graph instances.
+
+**REINDEX**
+:   Re-builds the index from the ground up using the data stored in the graph.
+Depending on the size of the graph, this action can take a long time to complete.
+Reindexing is possible in all stable states and automatically enables the index once finished.
+
+**ENABLE_INDEX**
+:   Enables the index so that it can be used by the query processing engine.
+An index must be registered before it can be enabled.
+If enabling the index manually instead of performing a reindex, be aware that past modifications of the graph are not represented in the index.
+Enabling a previously disabled index without `REINDEX` may cause index queries to return ghost vertices, which have been deleted from the graph while the index was disabled.
+
+**DISABLE_INDEX**
+:   Disables the index temporarily so that it is no longer used to answer queries.
```

Comment:
Why is `DISABLE_INDEX` from `ENABLED` to `DISABLED` not in grey, but `ENABLE_INDEX` from `DISABLED` to `ENABLED` in grey?

Reply:
The idea behind the grey transitions was to give an intuition for consequences to be aware of. Disabling an index does exactly what everybody probably expects, but plain enabling without reindexing could lead to faulty index responses. One has to keep in mind that ENABLE does not just reverse a prior DISABLE and that may not be obvious to everybody at first glance.